tests/gp_tests/test_batched_lbfgsb.py

```diff
@@ -0,0 +1,149 @@
+from scipy.optimize import fmin_l_bfgs_b
+
+from optuna._gp.batched_lbfgsb import batched_lbfgsb
+
+
+def rastrigin_and_grad(x: np.ndarray, batch_indices: np.ndarray) -> tuple[np.ndarray, np.ndarray]:
+    if x.ndim == 1:
+        x = x[None]
+    A = 10.0
+    dim = x.shape[-1]
+    _2pi_x = 2 * np.pi * x
+
+    fval = A * dim + np.sum(x**2 - A * np.cos(_2pi_x), axis=-1)
+    grad = 2 * x + 2 * np.pi * A * np.sin(_2pi_x)
+    return fval, grad
+
+
+def styblinski_tang_and_grad(
+    x: np.ndarray, batch_indices: np.ndarray
+) -> tuple[np.ndarray, np.ndarray]:
+    # Styblinski-Tang function, which has multiple local minima.
+    if x.ndim == 1:
+        x = x[None]
+    fval = np.sum(x**4 - 16 * x**2 + 5 * x, axis=-1) / 2
+    grad = 2 * x**3 - 16 * x + 2.5
+    return fval, grad
+
+
+def X0_and_bounds(dim: int, n_localopts: int) -> tuple[np.ndarray, np.ndarray]:
+    R = 5.12
+    rng = np.random.RandomState(0)
+    X0 = rng.random((n_localopts, dim)) * 2 * R - R
+    bounds = np.array([[-R, R]] * dim)
+    return X0, bounds
+
+
+def _verify_results(
+    X0: np.ndarray,
+    func_and_grad: Callable,
+    kwargs_ours: Any,
+    kwargs_scipy: Any,
+    batched_lbfgsb_func: Callable | None = None,
+) -> None:
+    if batched_lbfgsb_func is None:
+        batched_lbfgsb_func = batched_lbfgsb
+    xs_opt1, fvals_opt1, n_iters1 = batched_lbfgsb_func(
+        func_and_grad=func_and_grad, x0_batched=X0, **kwargs_ours
+    )
+    xs_opt2 = []
+    fvals_opt2 = []
+    n_iters2 = []
+    batch_indices = np.array([])
+    for x0 in X0:
+        x_opt, fval, info = fmin_l_bfgs_b(
+            func=func_and_grad, args=(batch_indices,), x0=x0, **kwargs_scipy
+        )
+        xs_opt2.append(x_opt)
+        fvals_opt2.append(float(fval))
+        n_iters2.append(info["nit"])
+
+    assert np.all(n_iters1 == np.array(n_iters2))
+    assert np.all(fvals_opt1 == np.array(fvals_opt2))
+    assert np.all(xs_opt1 == np.array(xs_opt2))
+
+
+test_params = [
+    (rastrigin_and_grad, {}, {}),
+    (styblinski_tang_and_grad, {}, {}),
+    (rastrigin_and_grad, {"max_evals": 3}, {"maxfun": 3}),
+    (styblinski_tang_and_grad, {"max_evals": 3}, {"maxfun": 3}),
+    (rastrigin_and_grad, {"max_iters": 3}, {"maxiter": 3}),
+    (styblinski_tang_and_grad, {"max_iters": 3}, {"maxiter": 3}),
+]
+
+
+@pytest.mark.parametrize("func_and_grad,kwargs_ours,kwargs_scipy", test_params)
+@pytest.mark.parametrize("use_greenlet", [True, False])
+@pytest.mark.parametrize("dim, n_localopts", [(10, 10), (1, 10), (10, 1), (1, 1)])
+def test_batched_lbfgsb(
+    monkeypatch: pytest.MonkeyPatch,
+    func_and_grad: Callable,
+    kwargs_ours: Any,
+    kwargs_scipy: Any,
+    use_greenlet: bool,
+    dim: int,
+    n_localopts: int,
+) -> None:
+    if not use_greenlet:
+        monkeypatch.setitem(sys.modules, "greenlet", None)
+
+    import optuna._gp.batched_lbfgsb as my_module
+
+    importlib.reload(my_module)
+    assert my_module._greenlet_imports.is_successful() is use_greenlet
```

Comment:
Please use `==` to check the value validity.

```suggestion
    assert my_module._greenlet_imports.is_successful() == use_greenlet
```